compiler/src/dotty/tools/dotc/typer/Synthesizer.scala

```diff
@@ -375,14 +375,130 @@ class Synthesizer(typer: Typer)(using @constructorOnly c: Context):
           synthesizedSumMirror(formal, span)
       case _ => EmptyTree
 
+  private def escapeJavaArray(elemTp: Type)(using Context): Type = elemTp match
+    case JavaArrayType(elemTp1) => defn.ArrayOf(escapeJavaArray(elemTp1))
+    case _ => elemTp
```

Comment:
Seems like this method could be used to solve https://github.com/lampepfl/dotty/issues/13131 :)